state/state_test.go

```diff
@@ -676,3 +671,19 @@ func TestCalculatingGenesisState(t *testing.T) {
 	r := tState1.calculateGenesisStateRootFromGenesisDoc()
 	assert.Equal(t, tState1.stateRoot(), r)
 }
+
+func TestCommittingInvalidBlock(t *testing.T) {
+	setup(t)
+
+	moveToNextHeightForAllStates(t)
+
+	txs := block.NewTxs()
+	trx := tState2.createSubsidyTx(crypto.GenerateTestAddress(), 0)
+	txs.Append(trx)
+	b := block.MakeBlock(2, util.Now(), txs, tState2.lastInfo.BlockHash(), tState2.stateRoot(), tState2.lastInfo.Certificate(), tState2.lastInfo.SortitionSeed(), tState2.signers[0].Address())
+	c := makeCertificateAndSign(t, b.Hash(), 0, tValSigner1, tValSigner2, tValSigner3, tValSigner4)
+
+	// If tState1 receives a block with version 2 and rejects it,
```

Comment:
// tState1 receives a block with version 2 and rejects it.
	// It is possible that the same block would be considered valid by tState2.